### PR TITLE
passport & jwt

### PR DESCRIPTION
**Pull request by chidieberejoel**

### CODE REVIEW

File: api/lib/utils.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import jsonwebtoken from "jsonwebtoken";
+import config from "../config";
+
+const { PRIV_KEY } = config;
+
+function issueJWT(user) {
```
Comment (tuzzy08) [inline api/lib/utils.js:6]: We can destructure the id in the parameters  --> function issueJWT({ _id }) and use directly in the payload object.

File: api/services/passport-local.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import UserDb from "../models/user";
+import signupValidator from "../middleware/validator";
+
+const authFields = {
+  usernameField: "email",
+  passwordField: "password",
+  passReqToCallback: true,
+};
+
+passport.serializeUser((user, done) => {
```
Comment (tuzzy08) [inline api/services/passport-local.js:14]: We are not managing sessions on the server and so we don't need to
serialize/deserialize user. Those functions saves the user id against the session, and should be removed.

File: api/services/passport-local.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  done(null, user.id);
+});
+
+passport.deserializeUser(async (id, done) => {
```
Comment (tuzzy08) [inline api/services/passport-local.js:18]: We are not managing sessions on the server and so we don't need to
serialize/deserialize user.

File: api/services/passport-local.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  })
+);
+
+passport.use(
+  "signup",
+  new Strategy(authFields, async (req, email, password, cb) => {
+    const salt = await bcrypt.genSalt(10);
+    const hashPassword = await bcrypt.hash(req.body.password, salt);
+    // enter param
+    const newUser = new UserDb({
```
Comment (tuzzy08) [inline api/services/passport-local.js:54]: You should create a new user object only after validation is successful, not before you check. So the flow is check for validation errors, then check if user exists after all that you can create a user object and save to db.

**Review by eagoyi — COMMENTED** [review]
To clear merge conflict, reset commit, update your branch to the develop branch, then update your work on the following existing folder, follow same process, while ensuring that on signup endpoint, email and password fields are validated with joi, password is hashed:
api/controller/userController.js
api/route/userRoute.js
index.js

**Comment by tuzzy08** [comment]
Please check

**Pull request closed without merging**